src/Messages/TSCall.m

```diff
@@ -86,22 +86,24 @@ - (BOOL)shouldAffectUnreadCounts
     return YES;
 }
 
-- (void)markAsReadLocallyWithTransaction:(YapDatabaseReadWriteTransaction *)transaction
+- (void)markAsReadWithTransaction:(YapDatabaseReadWriteTransaction *)transaction
+                  sendReadReceipt:(BOOL)sendReadReceipt
+                 updateExpiration:(BOOL)updateExpiration
 {
-    DDLogInfo(@"%@ marking as read uniqueId: %@ which has timestamp: %llu", self.tag, self.uniqueId, self.timestamp);
+    OWSAssert(transaction);
+
+    if (_read) {
+        return;
+    }
 
+    DDLogInfo(@"%@ marking as read uniqueId: %@ which has timestamp: %llu", self.tag, self.uniqueId, self.timestamp);
     _read = YES;
     [self saveWithTransaction:transaction];
-
-    // redraw any thread-related unread count UI.
     [self touchThreadWithTransaction:transaction];
-}
 
-- (void)markAsReadLocally
-{
-    [self.dbConnection readWriteWithBlock:^(YapDatabaseReadWriteTransaction *_Nonnull transaction) {
-        [self markAsReadLocallyWithTransaction:transaction];
-    }];
+    // Ignore sendReadReceipt and updateExpiration; they don't apply to calls.
+    //
+    // TODO: Should we update expiration of calls?
```

Comment:
Good question. Eventually, I think so.

Seems like it will require a little UI work since the current disappearing timer UI is predicated on the bubble-footer context, and has never been applied to a "system message" style interaction.